src/cli/conio.cpp

```diff
@@ -0,0 +1,93 @@
+/**
+ * conio.cpp
+ * 
+ * Particle serial input and output for command line use
+ * 
+*/
+
+
+#include "conio.hpp"
+
+#include "Particle.h"
+
+#include <cstdio>
+
+char SF_OSAL_printfBuffer[SF_OSAL_PRINTF_BUFLEN];
+
+
+
+extern "C"
+{
+    // Determines if key has been pressed
+    int kbhit(void) 
+    {
+        return Serial.available();
```

Comment:
It might not be a bad idea in the future to make `Serial` a compile-time define, that way it is trivial to switch to another Serial object.  That said, it is mostly moot for this effort.

Reply:
Sounds good